codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/generators/error/ErrorGenerator.kt

```diff
@@ -82,8 +82,14 @@ class ErrorGenerator(
     private fun renderError(isServer: Boolean) {
         val symbol = symbolProvider.toSymbol(shape)
         val messageShape = shape.errorMessageMember()
-        val message = messageShape?.let { "self.${symbolProvider.toMemberName(it)}.as_deref()" } ?: "None"
         val errorKindT = RuntimeType.errorKind(symbolProvider.config().runtimeConfig)
+        val (returnType, message) = messageShape?.let {
+            if (symbolProvider.config().handleRequired && messageShape.isRequired()) {
```

Comment:
I think this can use the same pattern of `symbolProvider.toSymbol(messageShape).isOptional()`

Reply:
It looks to be working, I probably had some leftover that a good clean removed..